### PR TITLE
Logic questions integration

### PR DESCRIPTION
**Pull request by Morpheu5**
- [x] Supports available symbols (letters only)
- [x] Exam boards syntax support
- [x] `/equality`

### CODE REVIEW

File: src/app/components/elements/InequalityModal.tsx [hunk 1]
```diff
@@ -93,9 +96,12 @@ export class InequalityModal extends React.Component<InequalityModalProps> {
...
-            console.log("New editor state: ", s);
+            const modal = document.getElementById('inequality-modal');
+            if (modal) {
+                this.setState({ editorState: s });
+                this.props.onEditorStateChange(s);
+                console.log("New editor state: ", s);
+            }
         };
         sketch.onCloseMenus = () => { this.setState({ menuOpen: false }) }; // TODO Maybe nice to have
         sketch.isUserPrivileged = () => true; // TODO Integrate with currentUser object
```
Comment (mlt47) [inline src/app/components/elements/InequalityModal.tsx]: Is user privileged important now? We can have access to the user form connecting the component to the Redux state store.

File: src/app/components/elements/InequalityModal.tsx [hunk 2]
```diff
@@ -183,50 +189,54 @@ export class InequalityModal extends React.Component<InequalityModalProps> {
...
+        let menu: JSX.Element =
+        <nav className="inequality-ui">
+            <div className={"inequality-ui menu-bar" + (this.state.menuOpen ? " open" : " closed")}>
+                {/* {this.state.activeMenu == "numbers" && <div className="top-menu numbers">
+                    <ul className="sub-menu">{
+                        "0123456789".split("").map((n, i) => {
+                            return this.menuItem(new MenuItem("Num", { significand: n }, { label: n, texLabel: true}), i);
+                        })
+                    }</ul>
+                </div>} */}
```
Comment (mlt47) [inline src/app/components/elements/InequalityModal.tsx]: Might want to clear up commented out code and/or any remaining console logs.

File: src/app/services/examBoard.ts [hunk 3]
```diff
@@ -1,8 +1,12 @@
 import {EXAM_BOARD} from "./constants";
 import {UserPreferencesDTO} from "../../IsaacAppTypes";
+import { user } from "../state/reducers";
 
 export const determineExamBoardFrom = (userPreferences: UserPreferencesDTO | null) => {
-    return (userPreferences && userPreferences.EXAM_BOARD && userPreferences.EXAM_BOARD.AQA) ?
-        EXAM_BOARD.AQA :
-        EXAM_BOARD.OCR
+    if (!userPreferences) {
```
Comment (mlt47) [inline src/app/services/examBoard.ts]: Would something like this be clearer?
``` java
    if (userPreferences && userPreferences.EXAM_BOARD && userPreferences.EXAM_BOARD.AQA) {
        return EXAM_BOARD.AQA;
    } else {
        return EXAM_BOARD.OCR;
    }
```

**Comment by jsharkey13** [comment]
@Morpheu5 - If you can make the changes Meurig suggested, and fix the merge conflict, I think this is good to go in.

File: src/app/components/elements/InequalityModal.tsx (hunk 1 above)
Comment (Morpheu5) [reply src/app/components/elements/InequalityModal.tsx]: I don't think it's relevant to CS, it was relevant to Physics for derivatives and differentials, which is why I'm setting it to true — but could have set it to false just as well.

File: src/app/services/examBoard.ts (hunk 3 above)
Comment (Morpheu5) [reply src/app/services/examBoard.ts]: Maybe? As long as it selects OCR by default is nothing else is selected…

**Comment by Morpheu5** [comment]
@jsharkey13 remind me tomorrow if I forget. This should be 15 minutes tops.

**Pull request merged**